src/Nethermind/Nethermind.Db/Metrics.cs

```diff
@@ -12,23 +12,76 @@ public static class Metrics
     {
         [CounterMetric]
         [Description("Number of Code DB cache reads.")]
-        public static long CodeDbCache { get; set; }
+        public static long CodeDbCache
+        {
+            get
+            {
+                long total = 0;
+                foreach (var value in _codeDbCache.Values)
+                {
+                    total += value;
+                }
+                return total;
+            }
+        }
+        private static ThreadLocal<long> _codeDbCache = new(trackAllValues: true);
+        [Description("Number of Code DB cache reads on thread.")]
+        public static long ThreadLocalCodeDbCache => _codeDbCache.Value;
+        public static void IncrementCodeDbCache() => _codeDbCache.Value++;
 
         [CounterMetric]
         [Description("Number of State Trie cache hits.")]
-        public static long StateTreeCache => _stateTreeCacheHits;
-        private static long _stateTreeCacheHits;
-        public static void IncrementTreeCacheHits() => Interlocked.Increment(ref _stateTreeCacheHits);
+        public static long StateTreeCache
+        {
+            get
+            {
+                long total = 0;
+                foreach (var value in _stateTreeCacheHits.Values)
+                {
+                    total += value;
+                }
+                return total;
+            }
+        }
+
+        private static ThreadLocal<long> _stateTreeCacheHits = new(trackAllValues: true);
+        public static void IncrementStateTreeCacheHits() => _stateTreeCacheHits.Value++;
```

Comment:
```suggestion
        public static long StateTreeCache
        {
            get
            {
                long total = 0;
                foreach (var value in _stateTreeCacheHits.Values)
                {
                    total += value;
                }
                return total;
            }
            set
            {
                _stateTreeCacheHits.Value = value;
            }
        }

        private static ThreadLocal<long> _stateTreeCacheHits = new(trackAllValues: true);
```

Reply:
getter would return the sum of all thread values; would add one to it; then set the current thread's value to the total+1

So it would start exponentially increasing

Reply:
Yeah would need to add `StateTreeCacheLocal` with a proper getter too